core/meterer/onchain_state.go

```diff
@@ -107,6 +109,33 @@ func (pcs *OnchainPaymentState) RefreshOnchainPaymentState(ctx context.Context,
 	}
 	// These parameters should be rarely updated, but we refresh them anyway
 	pcs.PaymentVaultParams = paymentVaultParams
```

Comment:
does this need locking?

Reply:
hmmm the case I would be concerned about is this gets updated in the middle of an on-demand payment, where it passes the old price per symbol but fails the new global rate. But with payment rollback, price per symbol isn't used. So far, I don't think we need locking and this seems atomic to me, but I'm happy to update if there's something I'm not seeing

Reply:
I don't think atomicity is guaranteed for pointer assignment. 
Can we use `atomic.Pointer`?

Reply:
I attempted this, lmk if that looks like what you are thinking